multicluster/controllers/multicluster/member/gateway_controller.go

```diff
@@ -176,7 +192,9 @@ func (r *GatewayReconciler) createResourceExport(ctx context.Context, req ctrl.R
 // SetupWithManager sets up the controller with the Manager.
 func (r *GatewayReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
-		For(&mcsv1alpha1.Gateway{}).
+		For(&mcv1alpha1.Gateway{}).
+		Watches(&source.Kind{Type: &mcv1alpha2.ClusterSet{}}, handler.EnqueueRequestsFromMapFunc(r.clusterSetMapFunc),
```

Comment:
Again, probably let ClusterSet controller post an event to other Reconcilers.

Reply:
I thought it before, but for reconciler process, I feel we'd rely on event mapping, the channel way is suitable for a one time case. But for reconcile, we are mapping multiple objects to one reconcile process, it's not suitable to use channel way since we need to handle every request's retry if there is any error. It's better to rely on controller runtime framework to handle the requests and retry.

Reply:
I did not get what is the difference. Even you trigger from ClusterSet controller, you can post an event for each affected object. The key is why we should depend on a remote object and another implicit relation, when you have all state in the local process.

Reply:
I got your point about the state are all in the local process. But in multi-cluster, we are not handling events for each objects directly, we rely on the controller framework to do reconcile. It will be triggered with event filter in `SetupWithManager(mgr ctrl.Manager)`, the framework will generate `[]reconcile.Request{}` for each object and call `Reconcile(ctx context.Context, req ctrl.Request)` correspondingly. 
I didn't get how to post an event for each affected object. Do you mean to generate `[]reconcile.Request{}` for each object? If so, it means we need to call `Reconcile(ctx context.Context, req ctrl.Request)` by ourselves per my understanding. It seems repeat the controller-runtime framework and not doable.

Reply:
Yeah, I think it's probably doable considering importer part is started after common area is ready. I will check how to refine this, but it will be a big change for current design, I doubt it can be done in this PR.

Reply:
Ok, we can think about a follow-up PR in the next release.